packages/sdk/lib/base.ts

```diff
@@ -75,12 +75,47 @@ export interface Logger {
   error(message: string): void;
 }
 
+export class PluginLogger implements Logger {
+  // eslint-disable-next-line no-useless-constructor,no-empty-function
+  constructor(private readonly plugin: Plugin, private readonly logger: Logger) {
+  }
+
+  logRequest(action: string, requestData?: string) {
+    const requestId = +new Date();
+    this.debug(`${this.plugin.id}: ${action}(request) ${requestId}: ${requestData || ''}`);
+    return {
+      success: (data?: string) => this.debug(`${this.plugin.id}: ${action}(response) ${requestId}: ${data || ''}`),
+      error: (data?: string) => this.error(`${this.plugin.id}: ${action}(response) ${requestId}: ${data || ''}`),
+    };
+  }
+
+  debug(message: string): void {
+    this.logger.debug(message);
+  }
+
+  error(message: string): void {
+    this.logger.error(message);
+  }
+
+  info(message: string): void {
+    this.logger.info(message);
+  }
+
+  warn(message: string): void {
+    this.logger.warn(message);
+  }
+}
+
 export abstract class Plugin {
+  protected logger: PluginLogger | undefined;
```

Comment:
I'd prefer we don't update the base Plugin class and keep it as lean as possible. What if move this into a separate plugin module This keeps the logic centralized and messages consistent but removes changes to the core SDK. I made a example here:
https://codesandbox.io/s/confident-sky-4tgdy?file=/src/itly/plugin-request-logger/RequestLoggerPlugin.ts

Reply:
Done.
I think:
1. `RequestLoggerPlugin` should a mixin instead of abstract base class. Maybe we will need to combine several behaviors.
2. The former `PluginLogger` could be used to add plugin-id prefixes in all 4 log methods